workable access to owner experiment detail from search activity

Ricky-Au opened the pull request:
This PR includes:
1) access to owner experiment detail screen from clicking an experiment after searching 
-Only passes Description though
2) framework for owner and non owner experiment detail screen for all buttons, all editTextViews and most(see known issues) textViews. 
-Only tested manually though

Known Issues:
1)Passing experiment object through intent always crashes app. Tried Serializable, Parcelable and some hacky bundle methods. 
- Suggestion: Since passing strings work without issue maybe we can have database functions to get experiment object by some unique experiment id.

2)For Owner details: 
- Description and Minimum trials EditTextViews automatic saving crashes app if edited twice.
    - Suggestion: We could just have a manual save button that updates these to the database for us
- Need database methods for unpublishing an experiment and ending experiment unless that is just deleting it.

3)For NonOwner details:
- Need  a "status" Field  added to the experiment class

4)For Both Owner and NonOwner details:
- In the storyboard there wasn't a good way to decide which screen to go to without checking if owner of experiment
  - Suggestion: Always go to NonOwner screen and have an edit button that only works if user is owner of the experiment
- Need methods to get standard deviation, median, mean and quartiles 
- Bottom navigation bar does not switch screens properly.
  - Suggestion: Remove navigation bars and have back buttons going back to search <- NonOwnerScreen <- OwnerScreen (if applicable)

vasu-gupta01 commented:
@Ricky-Au I decided to remove the bottom nav from anything that's not in the main activity, I think it might be better that way. 

For 4) there is a isOwner() method I implemented in Experiment class in may latest PR that might help. 

For 1) I was just doing this by passing the experiment ID to the intent and using getExperimentByID() which I also implemented in the Database class in my latest PR

Ricky-Au commented:
> @Ricky-Au I decided to remove the bottom nav from anything that's not in the main activity, I think it might be better that way.
> 
> For 4) there is a isOwner() method I implemented in Experiment class in may latest PR that might help.
> 
> For 1) I was just doing this by passing the experiment ID String to the intent and using getExperimentByID() which I also implemented in the Database class in my latest PR

Yup I saw those changes in your PR. That should fix those issues which is great.

arrooney commented on app/src/main/java/com/T05/krowdtrialz/ui/ExperimentDetailsNonOwnerActivity.java:
```diff
@@ -42,407 +62,527 @@
-            Hashtable<String, ArrayList<Float>> dateValues = new Hashtable<String, ArrayList<Float>>();
-            for (MeasurementTrial trial : measurementTrials){
-                String dateOfTrial = encodeDate(trial.getDateCreated().getYear(), trial.getDateCreated().getMonthValue(), trial.getDateCreated().getDayOfMonth());
-                ArrayList<Float> values;
-
-                if (dateValues.containsKey(dateOfTrial)){
-                    values = dateValues.get(dateOfTrial);
-                }else{
-                    values = new ArrayList<Float>();
-                }
-                values.add(trial.getMeasurementValue());
-                dateValues.put(dateOfTrial, values);
-            }
-
-            // make list of entries
-            for (String key : dateValues.keySet()) {
-                // turn your data into Entry objects
-                for (float value : dateValues.get(key)){
-                    entries.add(new Entry(decodeDate(key), value));
-                }
-
-            }
-
-            ScatterDataSet measurementDataSet = new ScatterDataSet(entries, measurementExperiment.getUnit());
-
-            measurementDataSet.setColors(new int[] {R.color.purple_700});
-
-            List<IScatterDataSet> dataSets = new ArrayList<IScatterDataSet>();
-
-            dataSets.add(measurementDataSet);
-
-
-            // create data set
-            ScatterData scatterData = new ScatterData(dataSets);
-
-            ValueFormatter formatter = new ValueFormatter() {
-                @Override
-                public String getAxisLabel(float value, AxisBase axis) {
-                    return formatDate(value);
-                }
```
Why is this being removed?

arrooney commented on app/src/main/java/com/T05/krowdtrialz/ui/ExperimentDetailsNonOwnerActivity.java:
```diff
@@ -42,407 +62,527 @@
+//                    values = new ArrayList<Integer>();
+//                }
+//                values.add(trial.getValue());
+//                dateValues.put(dateOfTrial, values);
+//            }
+//
+//            // make list of entries
+//            for (String key : dateValues.keySet()) {
+//                // turn your data into Entry objects
+//                for (int value : dateValues.get(key)){
+//                    entries.add(new Entry(decodeDate(key), value));
+//                }
+//
+//            }
+//
+//            ScatterDataSet integerDataSet = new ScatterDataSet(entries, integerExperiment.getUnit());
+//
+//            integerDataSet.setColors(new int[] {R.color.purple_700});
+//
+//            List<IScatterDataSet> dataSets = new ArrayList<IScatterDataSet>();
+//
+//            dataSets.add(integerDataSet);
+//
+//
+//            // create data set
+//            ScatterData scatterData = new ScatterData(dataSets);
+//
+//            ValueFormatter formatter = new ValueFormatter() {
+//                @Override
+//                public String getAxisLabel(float value, AxisBase axis) {
+//                    return formatDate(value);
+//                }
+//            };
+//            XAxis xAxis = scatterChart.getXAxis();
+//            xAxis.setGranularity(1f); // minimum axis-step (interval) is 1
+//            xAxis.setValueFormatter(formatter);
+//
+//            // add data to chart
+//            scatterChart.setData(scatterData);
```
Why is this commented out?

Jacques857 commented on app/src/main/java/com/T05/krowdtrialz/ui/ExperimentDetailsNonOwnerActivity.java:
```diff
@@ -44,15 +60,121 @@
+            }
+        });
+
+        populateMainInfo();
```
I believe these 4 lines should be in onSuccess. This is because the only time we have a guarantee that the query getExperimentByID has completed is once we are in onSuccess. Simply putting these lines after the listener is created does not guarantee the query is complete, and will likely result in null reference errors.
